src/freenet/client/filter/HTMLFilter.java

```diff
@@ -2111,14 +2112,19 @@ static class TagVerifier {
 		private final HashSet<String> uriAttrs;
 		private final HashSet<String> inlineURIAttrs;
 		final HashSet<String> booleanAttrs;
+		private final HashSet<String> allowedRole;
 
 		TagVerifier(String tag, String[] allowedAttrs) {
 			this(tag, allowedAttrs, null, null, null);
 		}
 
 		TagVerifier(String tag, String[] allowedAttrs, String[] uriAttrs, String[] inlineURIAttrs, String[] booleanAttrs) {
 			this.tag = tag;
-			this.allowedAttrs = new HashSet<String>();
+			this.allowedAttrs = new HashSet<String>(Arrays.asList("aria-braillelabel", "aria-brailleroledescription",
+				"aria-describedby", "aria-description", "aria-details", "aria-disabled", "aria-errormessage",
+				"aria-expanded", "aria-flowto","aria-hidden","aria-invalid","aria-label","aria-labelledby","aria-level",
+				"aria-multiline","aria-multiselectable","aria-orientation","aria-owns","aria-placeholder","aria-required",
+				"aria-roledescription","aria-selected","aria-setsize","aria-sort","aria-valuemax","aria-valuemin","aria-valuenow","aria-valuetext"));
```

Comment:
These allow different types of content —  boolean, free text, id, ... — so they need different validation.